Content.Shared/Xenoarchaeology/Artifact/SharedXenoArtifactSystem.XAE.cs

```diff
@@ -0,0 +1,99 @@
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Timing;
+using Content.Shared.Xenoarchaeology.Artifact.Components;
+using Robust.Shared.Map;
+
+namespace Content.Shared.Xenoarchaeology.Artifact;
+
+public abstract partial class SharedXenoArtifactSystem
+{
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+
+    private void InitializeXAE()
+    {
+        SubscribeLocalEvent<XenoArtifactComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<XenoArtifactComponent, AfterInteractEvent>(OnAfterInteract);
+        SubscribeLocalEvent<XenoArtifactComponent, ActivateInWorldEvent>(OnActivateInWorld);
+    }
+
+    private void OnUseInHand(Entity<XenoArtifactComponent> ent, ref UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        args.Handled = TryActivateXenoArtifact(ent, args.User, args.User, Transform(args.User).Coordinates);
+    }
+
+    private void OnAfterInteract(Entity<XenoArtifactComponent> ent, ref AfterInteractEvent args)
+    {
+        if (args.Handled || !args.CanReach)
+            return;
+
+        args.Handled = TryActivateXenoArtifact(ent, args.User, args.Target, args.ClickLocation);
+    }
+
+    private void OnActivateInWorld(Entity<XenoArtifactComponent> ent, ref ActivateInWorldEvent args)
+    {
+        if (args.Handled || !args.Complex)
+            return;
+
+        args.Handled = TryActivateXenoArtifact(ent, args.User, args.Target, Transform(args.Target).Coordinates);
+    }
+
+    public bool TryActivateXenoArtifact(
+        Entity<XenoArtifactComponent> artifact,
+        EntityUid? user,
+        EntityUid? target,
+        EntityCoordinates coordinates)
+    {
+        if (artifact.Comp.Suppressed)
+            return false;
+
+        if (TryComp<UseDelayComponent>(artifact, out var delay) && !_useDelay.TryResetDelay((artifact, delay), true))
```

Comment:
need to re-check that this is not blocking stuff permanently (that UseDelayCompoent is removed after delay passed?)